lib/Dialect/AIEVec/Transforms/VectorToAIEVecConversions.cpp

```diff
@@ -3306,12 +3312,8 @@ static void configureAIEVecCommonLegalizations(ConversionTarget &target,
     if (!isa<IntegerType>(srcScalarType) || !isa<IntegerType>(dstScalarType))
       return true;
 
-    unsigned srcLaneSize = getVectorLaneSize(srcType);
-    unsigned dstLaneSize = getVectorLaneSize(dstType);
-    unsigned srcElWidth = srcScalarType.getIntOrFloatBitWidth();
-    unsigned dstElWidth = dstScalarType.getIntOrFloatBitWidth();
-    return srcLaneSize != 32 || (dstElWidth <= srcElWidth) ||
-           (dstLaneSize != srcLaneSize);
+    // For aie2, ExtSIOp is always illegal
```

Comment:
If `ExtSIOp` is always illegal, then it's better to mark it as illegal (as opposed to dynamically legal). If it's only illegal for AIE2, illegality should be marked in the AIE2 legalizations, as opposed to the common ones.